libbeat/common/transport/tlscommon/tls_config.go

```diff
@@ -248,6 +251,25 @@ func makeVerifyServerConnection(cfg *TLSConfig) func(tls.ConnectionState) error
 }
 
 func verifyCertsWithOpts(certs []*x509.Certificate, casha256 []string, opts x509.VerifyOptions) error {
+	// if casha256 is defined, ignore the root CAs verification and verify
+	// only the fingerprint
+	for _, hexFingerprint := range casha256 {
+		fingerprint, _ := hex.DecodeString(hexFingerprint)
+
+		for _, cert := range certs {
+			// Compute digest for each certificate.
+			digest := sha256.Sum256(cert.Raw)
+
+			// Provided fingerprint should match at least one certificate before we continue.
+			if bytes.Compare(digest[0:], fingerprint) == 0 {
+				return nil
+			}
+		}
+	}
+
+	// if CASha256 is not defined, or we could not validate the fingerprint,
+	// proceed with the normal certificate validation
+
```

Comment:
That's just a PoC to ensure the fingerprint can be validated, this is far from the final implementation.